src/transaction_builder.js

```diff
@@ -406,7 +406,7 @@ function buildInput (input, allowIncomplete) {
   if (scriptType === bscript.types.P2SH) {
     // We can remove this error later when we have a guarantee prepareInput
     // rejects unsignable scripts - it MUST be signable at this point.
-    if (P2SH.indexOf(input.redeemScriptType) === -1) {
+    if (P2SH.indexOf(input.redeemScriptType) === -1 && !allowIncomplete) {
```

Comment:
down in line 417, I think it should be

```
if (input.redeemScriptType) {
  scriptType = input.redeemScriptType
}
```
because otherwise `scriptType` will end up getting set to undefined in the case where there is no `input.redeemScriptType`.